mico-admin/src/app/service-detail/service-detail.component.ts

```diff
@@ -1,46 +1,112 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 
 import { ApiService } from '../api/api.service';
-import { ApiObject } from '../api/apiobject';
 import { Subscription } from 'rxjs';
 
+export interface Service {
+    name: string;
+}
+
 @Component({
     selector: 'mico-service-detail',
     templateUrl: './service-detail.component.html',
     styleUrls: ['./service-detail.component.css']
 })
-export class ServiceDetailComponent implements OnInit {
+export class ServiceDetailComponent implements OnInit, OnDestroy {
 
-    private serviceSubscription: Subscription;
-    private paramSubscription: Subscription;
+    private subService: Subscription;
+    private subParam: Subscription;
 
     constructor(
         private apiService: ApiService,
         private route: ActivatedRoute,
     ) { }
 
-    @Input() service: ApiObject;
-
-    id: number;
+    service: Service;
+    shortName: string;
+    selectedVersion;
+    versions: any = [];
 
     ngOnInit() {
-        this.paramSubscription = this.route.params.subscribe(params => {
-            this.update(parseInt(params['id'], 10));
+        this.subParam = this.route.params.subscribe(params => {
+
+            this.shortName = params['shortName'];
+            const version = params['version'];
+            this.update(this.shortName, version);
         });
     }
 
-    update(id) {
-        if (id === this.id) {
+    ngOnDestroy() {
+        if (this.subService != null) {
+            this.subService.unsubscribe();
+        }
+        if (this.subParam != null) {
+            this.subParam.unsubscribe();
+        }
+    }
+
+    update(shortName, givenVersion) {
+
+        if (this.selectedVersion === givenVersion && givenVersion != null) {
             return;
         }
 
-        if (this.serviceSubscription != null) {
-            this.serviceSubscription.unsubscribe();
+        if (this.subService != null) {
+            this.subService.unsubscribe();
         }
 
-        this.serviceSubscription = this.apiService.getServiceById(id)
-        .subscribe(service => this.service = service);
+        // get latest version
+
+        // TODO change url path according to the current version
```

Comment:
should not be done in this method but in the method that calls this one change the call to a router.navigte